dj/construction/extract.py

```diff
@@ -0,0 +1,541 @@
+"""
+Functions for building queries, from nodes or SQL.
```

Comment:
Copy/pasta... "Functions for extracting dependencies and validating node queries." ?

Reply:
Thanks! This file was monolithic and I've moved all the build stuff elsewhere and this PR turned to just extraction with this big function